WordPress/gutenberg
# Navigation Editor: show a warning when trying to save unsupported blocks

## Description
Navigation editor only processes a subset of blocks in the editor, silently omitting anything that's not supported.
It would be useful to log a warning and make any problems there apparent.
Fixes https://github.com/WordPress/gutenberg/issues/35045.

## How has this been tested?

1. Create or edit an existing menu in the Navigation Editor (you have to enable it in the settings).
2. Make sure that the menu contains some unsupported blocks.
I had trouble reproducing this, so I've just replaced
https://github.com/WordPress/gutenberg/blob/92bf5e63a7a008fd4088f310a30691f4d990de85/packages/edit-navigation/src/store/utils.js#L80
with
```javascript
[  ].includes(
```
to emulate unsupported blocks.
3. Open the menu in the Navigation editor. Try to save it.
4. You should see a similar message in your browser console:
![Screenshot 2021-10-27 at 18 45 01](https://user-images.githubusercontent.com/43744263/139113538-4e3abd7e-8052-4d12-ae26-73e8f9280d2f.png)

## Types of changes


New feature


## Checklist:
